frontside/lib/views/home/home.dart

```diff
@@ -1,6 +1,69 @@
+import 'dart:async';
+
 import 'package:flutter/material.dart';
 import 'package:frontside/parts/navbar.dart';
 
+import 'package:frontside/services/places/autocomplete_response.pb.dart';
+import 'package:frontside/services/places/places.dart';
+import 'package:frontside/utils/debouncer.dart';
+
+class PlacesSearchDelegate extends SearchDelegate<AutocompleteEntry?> {
+  final _debouncer = new Debouncer();
+
+  @override
+  Widget buildLeading(BuildContext context) {
+    return IconButton(
+      icon: Icon(Icons.arrow_back),
+      onPressed: () {
+        close(context, null);
+      },
+    );
+  }
+
+  Future<AutocompleteResponse> _getResults() async {
+    final Completer<AutocompleteResponse> _completer = new Completer();
+
+    _debouncer.debounce(() async {
+      _completer.complete(await getPlaces(query));
+    });
+
+    return _completer.future;
+  }
+
+  @override
+  Widget buildResults(BuildContext context) {
+    return FutureBuilder(
+        builder: (BuildContext context,
+            AsyncSnapshot<AutocompleteResponse> response) {
+          var places = response.data?.data ?? [];
+
+          return ListView.builder(
+            itemBuilder: (context, index) {
+              var place = places[index];
+
+              return ListTile(
+                title: Text(place.name),
+                subtitle: Text(place.address),
+                onTap: () {
+                  close(context, place);
+                },
+              );
+            },
+            itemCount: places.length,
+          );
+        },
+        future: _getResults());
```

Comment:
for more clearness all props should be in the top of first arguments then you add builder method or widget tree, this counts for itemCount and future

Reply:
4a0cadf